src/libstd/fs.rs

```diff
@@ -422,6 +422,11 @@ impl OpenOptions {
     /// This option, when true, means that writes will append to a file instead
     /// of overwriting previous contents.
     ///
+    /// Note that this isn't the same as `write` with the cursor starting at
+    /// the end of the file; this mode only allows writing to the end of the
```

Comment:
Yeah I think the append mode on files is definitely separate from this. I believe the confusion derives from the title of this PR (append doesn't imply write), but that just means that if you want to write to the file you also need to say `write(true)`. Perhaps that could be mentioned in these docs?